### PR TITLE
@types/stripe Added collection_method to IInvoiceCreationOptions

### PR DESCRIPTION
**Pull request by jvik**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api/invoices/object

---

I changed billing to collection_method as the API has renamed the attribute

### CODE REVIEW

**Comment by jvik** [comment]
In the spirit of Hacktoberfest I'm trying to fix issue #38835 myself

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -3472,6 +3473,8 @@ declare namespace Stripe {
              * Stripe will attempt to pay this invoice using the default source attached to the
              * customer. When sending an invoice, Stripe will email this invoice to the customer
              * with payment instructions.
+             *
+             * This field has been renamed to collection_method and will be removed in a future API version.
```
Comment (pushplay) [inline types/stripe/index.d.ts]: An `@deprecated` tag here would be a good idea.

File: types/stripe/index.d.ts [hunk 2]
```diff
@@ -3835,8 +3838,9 @@ declare namespace Stripe {
              * will attempt to pay this invoice using the default source attached to the customer.
              * When sending an invoice, Stripe will email this invoice to the customer with payment
              * instructions. Defaults to charge_automatically.
+             *
              */
-            billing?: 'charge_automatically' | 'send_invoice';
```
Comment (pushplay) [inline types/stripe/index.d.ts]: Is this a recent API change or should it always have been `collection_method`?  If it's an API change the old REST API is still usable with the latest library version so the old param should still be available but `@deprecated`.

**Review by pushplay — CHANGES_REQUESTED** [review]

File: types/stripe/index.d.ts (hunk 1 above)
Comment (jvik) [reply types/stripe/index.d.ts]: Fixed

File: types/stripe/index.d.ts (hunk 2 above)
Comment (jvik) [reply types/stripe/index.d.ts]: fixed

**Review by pushplay — APPROVED** [review]

**Comment by jvik** [comment]
Merge conflict fixed @pushplay

**Review by richardwardza — APPROVED** [review]

**Pull request merged**